src/ImageRunModal.jsx

```diff
@@ -351,68 +353,116 @@ export class ImageRunModal extends React.Component {
-
         if (this.activeConnection)
             this.activeConnection.close();
 
         this.setState({ searchFinished: false, searchInProgress: true });
         this.activeConnection = rest.connect(client.getAddress(this.isSystem()), this.isSystem());
-        let searches = [];
-
-        // If there are registries configured search in them, or if a user searches for `docker.io/cockpit` let
-        // podman search in the user specified registry.
-        if (Object.keys(this.props.podmanInfo.registries).length !== 0 || value.includes('/')) {
-            searches.push(this.activeConnection.call({
-                method: "GET",
-                path: client.VERSION + "libpod/images/search",
-                body: "",
-                params: {
-                    term: value,
-                }
-            }));
-        } else {
-            searches = searches.concat(utils.fallbackRegistries.map(registry =>
-                this.activeConnection.call({
+        const searches = [];
+
+        // Try to get specified image manifest
+        searches.push(this.activeConnection.call({
+            method: "GET",
+            path: client.VERSION + "libpod/manifests/" + value + "/json",
+            body: "",
+        }));
+
+        // Don't start search queries when tag is specified as search API doesn't support
+        // searching for specific image tags
+        // instead only rely on manifests query (requires image:tag name)
+        if (!RE_CONTAINER_TAG.test(value)) {
+            // If there are registries configured search in them, or if a user searches for `docker.io/cockpit` let
+            // podman search in the user specified registry.
+            if (Object.keys(this.props.podmanInfo.registries).length !== 0 || value.includes('/')) {
+                searches.push(this.activeConnection.call({
                     method: "GET",
                     path: client.VERSION + "libpod/images/search",
                     body: "",
                     params: {
-                        term: registry + "/" + value
+                        term: value,
                     }
-                })));
+                }));
+            } else {
+                searches.push(...utils.fallbackRegistries.map(registry =>
+                    this.activeConnection.call({
+                        method: "GET",
+                        path: client.VERSION + "libpod/images/search",
+                        body: "",
+                        params: {
+                            term: registry + "/" + value
+                        }
+                    })));
+            }
         }
 
         Promise.allSettled(searches)
                 .then(reply => {
                     if (reply && this._isMounted) {
                         let imageResults = [];
-                        let dialogError = "";
-                        let dialogErrorDetail = "";
+                        const manifestResult = handleManifestsQuery(reply[0]);
```

Comment:
So if this fails, but the other search replies succeed, this quickly sets an error in the dialog, but the code below will then reset/update the error, right? I.e. that may be intended (even though it's a bit ugly). The error handling strategy should at least be commented here, or better made explicit.

Reply:
I'm not sure if manifests errors should be reset by successful search queries? Since it looks for a specific image in a specific location and search queries are a bit more lax. Error 404 is handled separately and other errors from the 500s are unexpected so I believe that should be visible in this case.

Reply:
I re-read this again: changeDialogError() is only called by the search results if `!manifestResult`. So indeed if the manifest query fails, then `manifestResult === null`, and the error message *does* get overwritten by search failures. I agree that the manifest query error is worth showing, but that's not what the code is doing?

Reply:
So I suppose the way to go is to show the first encountered error

Reply:
Yeah, at this point all promises are settled, so you can consider all failures and know where they come from.

Reply:
Thanks, I'll stop picking on the error handling -- this is good enough now. Thanks!